### ISSUE TITLE
False positives for some methods of lists for boolean trap rule FBT003

### ISSUE THREAD

**Issue by namurphy** [charliermarsh/ruff#4382]
When providing `True` or `False` as an argument to the `append`, `count`, `insert`, and `remove` methods of a `list`, the rule for catching boolean traps (`FBT003`) yields false positives even though the arguments are positional-only.

I created `file.py` which contains:
```python
some_list = []
some_list.append(True)
some_list.count(True)
some_list.insert(0, True)
some_list.remove(True)
```
Using ruff `v0.0.265`, I ran:
```bash
ruff file.py --select FBT003 --isolated
```
and got the output:
```
file.py:2:18: FBT003 Boolean positional value in function call
file.py:3:17: FBT003 Boolean positional value in function call
file.py:4:21: FBT003 Boolean positional value in function call
file.py:5:18: FBT003 Boolean positional value in function call
```
This follows up on #1336.

Thank you!

**Comment by namurphy** [comment]
Wow, that was quick!  Thank you for fixing it!

**Comment by charliermarsh** [comment]
No prob, it's an easy one :)
